source/base/quadrature.cc

```diff
@@ -284,6 +295,39 @@ Quadrature<dim>::memory_consumption () const
 }
 
 
+
+template <int dim>
+Quadrature<1>
+Quadrature<dim>::get_tensor_basis () const
+{
+  Assert (this->is_tensor_product_flag == true,
+          ExcMessage("This function only makes sense if this object represents a tensor product!"));
+
+  // Just take the first components of the first quadrature points
+  // and rescale the weights accordingly.
+  const unsigned int n_q_points_1d = std::round(std::pow(this->size(), 1./dim));
+  std::vector<Point<1> > q_points_1d (n_q_points_1d);
+  std::vector<double> weights_1d (n_q_points_1d);
+
+  double sum = 0.;
+  for (unsigned int i=0; i<n_q_points_1d; ++i)
+    {
+      sum += this->weight(i);
+      q_points_1d[i](0) = this->point(i)(0);
+    }
+
+  if (!std::isinf(sum))
+    for (unsigned int i=0; i<n_q_points_1d; ++i)
+      weights_1d[i] = this->weight(i)/sum;
+  else
+    for (unsigned int i=0; i<n_q_points_1d; ++i)
+      weights_1d[i] = this->weight(i);
```

Comment:
The constructors that just takes a  `std::vector<Point<dim>` as argument initializes the corresponding weights with `std::atof("Inf")`.
Now, I also test for the sum to be larger than zero.